File: py/IEC61968/Metering/DemandResponseProgram.py
# Converted by an OPENAI API call using model: gpt-3.5-turbo-1106
from CIM_STD_PYTHON.TC57CIM.IEC61968.Customers.CustomerAgreement import CustomerAgreement
from CIM_STD_PYTHON.TC57CIM.IEC61970.Base.Core.IdentifiedObject import IdentifiedObject
from CIM_STD_PYTHON.TC57CIM.IEC61970.Base.Domain.DateTimeInterval import DateTimeInterval


class DemandResponseProgram(IdentifiedObject):
    
    def __init__(self):
        super().__init__()
        self.type = ""
        self.validity_interval = DateTimeInterval()
        self.usage_point_groups = None  # Type UsagePointGroup not defined
        self.end_device_groups = None    # Type EndDeviceGroup not defined
        self.customer_agreements = CustomerAgreement()

File: java/IEC61968/Metering/DemandResponseProgram.java
package IEC61968.Metering;

import IEC61970.Base.Domain.String;
import IEC61970.Base.Domain.DateTimeInterval;
import IEC61970.Base.Core.IdentifiedObject;
import IEC61968.Customers.CustomerAgreement;

/**
 * Demand response program.
 * @author T. Kostic
 * @version 1.0
 * @created 25-Dec-2023 8:45:20 PM
 */
public class DemandResponseProgram extends IdentifiedObject {

	/**
	 * Type of demand response program; examples are CPP (critical-peak pricing), RTP
	 * (real-time pricing), DLC (direct load control), DBP (demand bidding program),
	 * BIP (base interruptible program). Note that possible types change a lot and it
	 * would be impossible to enumerate them all.
	 */
	public String type;
	/**
	 * Interval within which the program is valid.
	 */
	public DateTimeInterval validityInterval;
	/**
	 * All usage point groups enrolled in this demand response program.
	 */
	public UsagePointGroup UsagePointGroups;
	/**
	 * All groups of end devices enrolled in this demand response program.
	 */
	public EndDeviceGroup EndDeviceGroups;
	/**
	 * All customer agreements through which the customer is enrolled in this demand
	 * response program.
	 */
	public CustomerAgreement CustomerAgreements;

	public DemandResponseProgram(){

	}
}//end DemandResponseProgram